common/lib/capa/capa/capa_problem.py

```diff
@@ -808,7 +822,13 @@ def _extract_html(self, problemtree):  # private
         # otherwise, render children recursively, and copy over attributes
         tree = etree.Element(problemtree.tag)
         for item in problemtree:
-            item_xhtml = self._extract_html(item)
+            item_xhtml = self._extract_html(item, question_id)
+
+            if item.tag == 'question':
+                item_xhtml.set('class', 'question')
+                item_xhtml.set('id', 'question-{}'.format(question_id))
```

Comment:
This ID is still not going to be unique if there is more than one problem on the page. You can see this by creating 2 problems in the same unit, then checking for accessibility issues in the browser (see screenshot below). An a11y bok choy test with 2 problems on the same page would also catch this.

I would recommend including the xblock ID, which is guaranteed to be unique.

![image](https://cloud.githubusercontent.com/assets/484484/15436699/481bc1fc-1e91-11e6-812c-c229535e4576.png)